src/main/java/com/ghilly/service/CountryServiceRest.java

```diff
@@ -1,43 +1,58 @@
 package com.ghilly.service;
 
-import org.slf4j.LoggerFactory;
+import com.ghilly.repository.CountryRepositoryRest;
 import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.util.ArrayList;
 import java.util.List;
 
+
 public class CountryServiceRest implements CountryService {
 
     private static final Logger logger = LoggerFactory.getLogger(CountryServiceRest.class);
+    private final CountryRepositoryRest repository;
+
+    public CountryServiceRest(CountryRepositoryRest repository) {
+        this.repository = repository;
+    }
 
 
     @Override
-    public String add(String countryName) {
+    public void add(String countryName) {
+        repository.insert(countryName);
         logger.info("The country {} was added by service.", countryName);
-        return countryName;
     }
 
     @Override
-    public List<String> getAll() {
-        logger.info("getAll");
-        return new ArrayList<>();
+    public List<String> getAllCountries() {
+        logger.info("The list of countries is:");
+        return repository.takeAllCountries();
     }
 
     @Override
     public String getCountry(int countryId) {
-        logger.info("getCountry");
-        return "There will be the country name";
+        if (!repository.containsCountry(countryId)) {
+            throw new IllegalArgumentException("The country is not found.");
+        }
+        return repository.takeCountry(countryId);
     }
 
     @Override
-    public String upgrade(int countryId, String newCountryName) {
-        logger.info("The country with ID {} was upgraded, new name is {}.", countryId, newCountryName);
-        return newCountryName;
+    public void upgrade(int countryId, String newName) {
+        if (!repository.containsCountry(countryId)) {
+            throw new IllegalArgumentException("The country is not found.");
```

Comment:
pleae add countyId, without id we don't recognize what country is not found

Reply:
+